src/callbacks_step/glm_speed.jl

```diff
@@ -0,0 +1,70 @@
+
+"""
+    GlmSpeedCallback(; glm_scale=0.5, cfl=1.0)
+
+Update the divergence cleaning wave speed `c_h` according to the time step
+computed in [`StepsizeCallback`](@ref) for the ideal GLM-MHD equations.
+"""
+struct GlmSpeedCallback{RealT<:Real}
+  glm_scale::RealT
+  cfl::RealT
+end
+
+
+function Base.show(io::IO, cb::DiscreteCallback{Condition,Affect!}) where {Condition, Affect!<:GlmSpeedCallback}
+  glm_speed_callback = cb.affect!
+  @unpack glm_scale, cfl = glm_speed_callback
+  print(io, "GlmSpeedCallback(glm_scale=", glm_scale, ", cfl=", cfl, ")")
+end
+
+
+function Base.show(io::IO, ::MIME"text/plain", cb::DiscreteCallback{Condition,Affect!}) where {Condition, Affect!<:GlmSpeedCallback}
+  if get(io, :compact, false)
+    show(io, cb)
+  else
+    glm_speed_callback = cb.affect!
+
+    setup = [
+             "GLM wave speed scaling" => glm_speed_callback.glm_scale,
+             "Expected CFL number" => glm_speed_callback.cfl,
+            ]
+    summary_box(io, "GlmSpeedCallback", setup)
+  end
+end
+
+
+function GlmSpeedCallback(; glm_scale=0.5, cfl=1.0)
```

Comment:
I reread the Dedner et al. paper and yes, the scaling should be in the interval [0,1] (otherwise it can alter the physical time step).

In my experience, no. The CFL scale should always match the scale of the physical problem. I have seen people use a a different CFL number for the cleaning such that they take several cleaning steps in-between physical steps (this is a sort of sub-cycling) but IIRC it such subcycling works well for linear probelms (like Maxwell's equations) but not very well for nonlinear problems (like ideal MHD).

Reply:
OK. Then I would definitely check for a valid value for the `glm_scale` and make the CFL a required keyword argument (i.e., without a default value).